Shopify/hydrogen
# ShopPayButton now accepts storeDomain prop

ShopPayButton can now receive a storeDomain prop. Updated the demo store to make it work.

### WHY are these changes introduced?

Part of: https://github.com/Shopify/hydrogen/issues/629
Solves:
https://github.com/Shopify/hydrogen/issues/487

### HOW to test your changes?
Go to a product in the demo store and click on Shop Pay button. It should open the Shop checkout

#### Post-merge steps
There are other component relying on `ShopifyProvider`

#### Checklist
